plugins/scaffolder-node/src/permissions/rules/scaffolderTemplateRules.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { makeCreatePermissionRule } from '@backstage/plugin-permission-node';
+import {
+  TemplateEntityStepV1beta3,
+  TemplateParametersV1beta3,
+} from '@backstage/plugin-scaffolder-common';
+import { RESOURCE_TYPE_SCAFFOLDER_TEMPLATE } from '@backstage/plugin-scaffolder-common/alpha';
+import { z } from 'zod';
+
+/**
+ * @public
+ */
+export const createTemplatePermissionRule = makeCreatePermissionRule<
+  TemplateEntityStepV1beta3 | TemplateParametersV1beta3,
+  {},
+  typeof RESOURCE_TYPE_SCAFFOLDER_TEMPLATE
+>();
+
+const hasTag = createTemplatePermissionRule({
```

Comment:
mmm what's the reason for moving all the rules over to `scaffolder-node`?

Reply:
~~it's to follow prior work in the Catalog plugin, where rules and conditions are in the `-node` module~~

Annnd... double checking that, turns out I misread something at some point and they are *actually* in the `-backend` module. I'll move them back.